### PR TITLE
feat(PubSub): Adds emulator support sample

### PR DESCRIPTION
**Pull request by amanda-tarafa**
And fixes some `async void` in a separate commit.

### CODE REVIEW

File: pubsub/api/Pubsub.Samples/EmulatorSupport.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        // variable is set, but will otherwise connect to the production environment.
+
+        // Create the PublisherServiceApiClient using the PublisherServiceApiClientBuilder
+        // and setting the EmulatorDection property.
+        PublisherServiceApiClient publisherService = await new PublisherServiceApiClientBuilder
+        {
+            EmulatorDetection = EmulatorDetection.EmulatorOrProduction
+        }.BuildAsync();
+
+        // Use the client as you'd normally do, to crate a topic in this example.
```
Comment (jskeet) [inline pubsub/api/Pubsub.Samples/EmulatorSupport.cs:38]: crate => create

File: pubsub/api/Pubsub.Samples/EmulatorSupport.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        publisherService.CreateTopic(topicName);
+
+        // Create the SubscriberServiceApiClient using the SubscriberServiceApiClientBuilder
+        // and setting the EmulatorDection property.
+        SubscriberServiceApiClient subscriberService = await new SubscriberServiceApiClientBuilder
+        {
+            EmulatorDetection = EmulatorDetection.EmulatorOrProduction
+        }.BuildAsync();
+
+        // Use the client as you'd normally do, to create a subscrition in this example.
```
Comment (jskeet) [inline pubsub/api/Pubsub.Samples/EmulatorSupport.cs:49]: subscrition => subscription

File: pubsub/api/Pubsub.Samples/EmulatorSupport.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        List<PubsubMessage> receivedMessages = new List<PubsubMessage>();
+
+        // Use the client as you'd normally do, to listen for messages in this example.
+        await subscriber.StartAsync((msg, cancellationToken) =>
+        {
+            receivedMessages.Add(msg);
+            Console.WriteLine($"Received message {msg.MessageId} published at {msg.PublishTime.ToDateTime()}");
+            Console.WriteLine($"Text: '{msg.Data.ToStringUtf8()}'");
+            // In this example we stop the subscriber when the message is received.
+            // You may leave the subscribir running, and it will continue to received published messages
```
Comment (jskeet) [inline pubsub/api/Pubsub.Samples/EmulatorSupport.cs:78]: subscribir => subscriber

**Review by jskeet — APPROVED** [review]
Just typos.

**Comment by amanda-tarafa** [comment]
Thanks! I'll wait for @anguillanneuf to take a look before merging. I'll fix the typos as well.

**Comment by amanda-tarafa** [comment]
I'm merging this one for now, @anguillanneuf when you are back let me know if you'd like me to make some changes. Thanks!

**Comment by anguillanneuf** [comment]
Thanks Amanda!

**Pull request merged**